packages/cli/src/utils/index.js

```diff
@@ -56,6 +56,10 @@ export async function loadNuxtConfig(argv) {
   options.mode =
     (argv.spa && 'spa') || (argv.universal && 'universal') || options.mode
 
+  if (options.mode && !['spa', 'universal'].includes(options.mode)) {
```

Comment:
The way that `mode` is designed is like a defined preset. We can add more modes using `options.modes`.  So i suggest moving this check after resolving `modePreset`: https://github.com/nuxt/nuxt.js/blob/dev/packages/config/src/options.js#L240

Reply:
I felt that I missed something like that. Thanks for the heads up! 🙈